### PR TITLE
[DX-1069] TT-11263 documented TYK_IB_SESSION_SECRET

### PR DESCRIPTION
**Pull request by sredxny**
### For internal users - Please add a Jira DX PR ticket to the subject!
<br>
https://tyktech.atlassian.net/browse/DX-1069
---

### Preview Link
[Preview Link](https://deploy-preview-4139--tyk-docs.netlify.app/docs/nightly/tyk-identity-broker/getting-started/#configuring-tyk_ib_session_secret)

### Description

<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [ ] I have added a **preview link** to the PR description.
- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I **labelled** the PR

___

## **Type**
documentation

___

## **Description**
- Documented the configuration of `TYK_IB_SESSION_SECRET` environment variable for Tyk Identity Broker (TIB).
- Highlighted the significance of `TYK_IB_SESSION_SECRET` in enhancing security by hashing session cookies.
- Included guidance on setting up `TYK_IB_SESSION_SECRET` on Linux, Unix, or MacOS systems.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>getting-started.md</strong><dd><code>Documenting `TYK_IB_SESSION_SECRET` Configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-identity-broker/getting-started.md
<li>Added documentation for configuring <code>TYK_IB_SESSION_SECRET</code> environment <br>variable.<br> <li> Explained the importance of <code>TYK_IB_SESSION_SECRET</code> for securing session <br>cookies.<br> <li> Provided instructions on how to set the <code>TYK_IB_SESSION_SECRET</code> variable <br>in Linux, Unix, or MacOS environments.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4139/files#diff-538a31e18e28a8d94aeeb3907b9463d55520749892dc23e39982f7a958c50cee">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table><hr>

<details> <summary><strong>✨ Usage guide:</strong></summary><hr> 

**Overview:**
The `describe` tool scans the PR code changes, and generates a description for the PR - title, type, summary, walkthrough and labels. The tool can be triggered [automatically](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#github-app-automatic-tools) every time a new PR is opened, or can be invoked manually by commenting on a PR.

When commenting, to edit [configurations](https://github.com/Codium-ai/pr-agent/blob/main/pr_agent/settings/configuration.toml#L46) related to the describe tool (`pr_description` section), use the following template:
```
/describe --pr_description.some_config1=... --pr_description.some_config2=...
```
With a [configuration file](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#working-with-github-app), use the following template:
```
[pr_description]
some_config1=...
some_config2=...
```

<table><tr><td><details> <summary><strong> Enabling\disabling automation </strong></summary><hr>

- When you first install the app, the [default mode](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#github-app-automatic-tools) for the describe tool is:
```
pr_commands = ["/describe --pr_description.add_original_user_description=true" 
                         "--pr_description.keep_original_user_title=true", ...]
```
meaning the `describe` tool will run automatically on every PR, will keep the original title, and will add the original user description above the generated description. 

- Markers are an alternative way to control the generated description, to give maximal control to the user. If you set:
```
pr_commands = ["/describe --pr_description.use_description_markers=true", ...]
```
the tool will replace every marker of the form `pr_agent:marker_name` in the PR description with the relevant content, where `marker_name` is one of the following:
  - `type`: the PR type.
  - `summary`: the PR summary.
  - `walkthrough`: the PR walkthrough.

Note that when markers are enabled, if the original PR description does not contain any markers, the tool will not alter the description at all.
        

</details></td></tr>

<tr><td><details> <summary><strong> Custom labels </strong></summary><hr>

The default labels of the `describe` tool are quite generic: [`Bug fix`, `Tests`, `Enhancement`, `Documentation`, `Other`].

If you specify [custom labels](https://github.com/Codium-ai/pr-agent/blob/main/docs/DESCRIBE.md#handle-custom-labels-from-the-repos-labels-page-gem) in the repo's labels page or via configuration file, you can get tailored labels for your use cases.
Examples for custom labels:
- `Main topic:performance` - pr_agent:The main topic of this PR is performance
- `New endpoint` - pr_agent:A new endpoint was added in this PR
- `SQL query` - pr_agent:A new SQL query was added in this PR
- `Dockerfile changes` - pr_agent:The PR contains changes in the Dockerfile
- ...

The list above is eclectic, and aims to give an idea of different possibilities. Define custom labels that are relevant for your repo and use cases.
Note that Labels are not mutually exclusive, so you can add multiple label categories.
Make sure to provide proper title, and a detailed and well-phrased description for each label, so the tool will know when to suggest it.        

</details></td></tr>

<tr><td><details> <summary><strong> Inline File Walkthrough 💎</strong></summary><hr>

For enhanced user experience, the `describe` tool can add file summaries directly to the "Files changed" tab in the PR page.
This will enable you to quickly understand the changes in each file, while reviewing the code changes (diffs).

To enable inline file summary, set `pr_description.inline_file_summary` in the configuration file, possible values are:
- `'table'`: File changes walkthrough table will be displayed on the top of the "Files changed" tab, in addition to the "Conversation" tab.
- `true`: A collapsable file comment with changes title and a changes summary for each file in the PR.
- `false` (default): File changes walkthrough will be added only to the "Conversation" tab.
<tr><td><details> <summary><strong> Utilizing extra instructions</strong></summary><hr>

The `describe` tool can be configured with extra instructions, to guide the model to a feedback tailored to the needs of your project.

Be specific, clear, and concise in the instructions. With extra instructions, you are the prompter. Notice that the general structure of the description is fixed, and cannot be changed. Extra instructions can change the content or style of each sub-section of the PR description.

Examples for extra instructions:
```
[pr_description] 
extra_instructions="""
- The PR title should be in the format: '<PR type>: <title>'
- The title should be short and concise (up to 10 words)
- ...
"""
```
Use triple quotes to write multi-line instructions. Use bullet points to make the instructions more readable.

</details></td></tr>

<tr><td><details> <summary><strong> More PR-Agent commands</strong></summary><hr> 

> To invoke the PR-Agent, add a comment using one of the following commands:  
> - **/review**: Request a review of your Pull Request.   
> - **/describe**: Update the PR title and description based on the contents of the PR.   
> - **/improve [--extended]**: Suggest code improvements. Extended mode provides a higher quality feedback.   
> - **/ask \<QUESTION\>**: Ask a question about the PR.   
> - **/update_changelog**: Update the changelog based on the PR's contents.   
> - **/add_docs** 💎: Generate docstring for new components introduced in the PR.   
> - **/generate_labels** 💎: Generate labels for the PR based on the PR's contents.   
> - **/analyze** 💎: Automatically analyzes the PR, and presents changes walkthrough for each component.   

>See the [tools guide](https://github.com/Codium-ai/pr-agent/blob/main/docs/TOOLS_GUIDE.md) for more details.
>To list the possible configuration parameters, add a **/config** comment.   
 

</details></td></tr>

</table>

See the [describe usage](https://github.com/Codium-ai/pr-agent/blob/main/docs/DESCRIBE.md) page for a comprehensive guide on using this tool.

</details>

### CODE REVIEW

File: tyk-docs/content/tyk-identity-broker/getting-started.md [hunk 1]
```diff
@@ -38,6 +38,11 @@ This settings behaves as follows:
 * If `enabled` = `true` and the tib host is not present the internal TIB will be loaded
 * If `enabled` = `true` and the tib host is set, then external TIB will be loaded
 
+### Configuring TYK_IB_SESSION_SECRET
+To secure session cookies within Tyk Identity Broker (TIB) when integrating with social providers, setting the `TYK_IB_SESSION_SECRET` environment variable is crucial. This variable plays a pivotal role in hashing session cookies, thereby enhancing security. By default, if this variable isn't explicitly set, TIB falls back to using the Tyk Dashboard's admin_secret when it's embedded in the dashboard.
+
+For a seamless and secure setup, start by generating a strong, unique secret string. This string will be your session secret. In a Linux, Unix, or MacOS environment, you can set this variable by running the command `export TYK_IB_SESSION_SECRET='your_secret'`.
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-identity-broker/getting-started.md]: Can you recommend to the reader for what makes a strong secret.

File: tyk-docs/content/tyk-identity-broker/getting-started.md [hunk 2]
```diff
@@ -38,6 +38,11 @@ This settings behaves as follows:
 * If `enabled` = `true` and the tib host is not present the internal TIB will be loaded
 * If `enabled` = `true` and the tib host is set, then external TIB will be loaded
 
+### Configuring TYK_IB_SESSION_SECRET
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-identity-broker/getting-started.md]: Is this a typo  TYK_IB_SESSION_SECRET? Should it be  TYK_TIB_SESSION_SECRET?

File: tyk-docs/content/tyk-identity-broker/getting-started.md [hunk 3]
```diff
@@ -38,6 +38,11 @@ This settings behaves as follows:
...
+To secure session cookies within Tyk Identity Broker (TIB) when integrating with social providers, setting the `TYK_IB_SESSION_SECRET` environment variable is crucial. This variable plays a pivotal role in hashing session cookies, thereby enhancing security. By default, if this variable isn't explicitly set, TIB falls back to using the Tyk Dashboard's admin_secret when it's embedded in the dashboard.
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-identity-broker/getting-started.md:42]: Is this a typo  TYK_IB_SESSION_SECRET? Should it be  TYK_TIB_SESSION_SECRET?

Does this mean that the dashboard admin secret could be leaked to a 3rd party Identity Provider? If so maybe explain this.

**Review by dcs3spp — CHANGES_REQUESTED** [review]
@sredxny PR looks good to me. Left comments:
- Checking if there is a typo
- Recommend to reader what makes a strong secret

File: tyk-docs/content/tyk-identity-broker/getting-started.md (hunk 2 above)
Comment (sredxny) [reply tyk-docs/content/tyk-identity-broker/getting-started.md]: @dcs3spp it's not a typo, the var is defined like that. It's the same when using embeded or standalone TIB

File: tyk-docs/content/tyk-identity-broker/getting-started.md (hunk 3 above)
Comment (sredxny) [reply tyk-docs/content/tyk-identity-broker/getting-started.md:42]: @dcs3spp good spot! am gathering info to know if it represents a security issue, so far I dont think so as in prod environments a lot of customers are already defining the configuration via env vars

File: tyk-docs/content/tyk-identity-broker/getting-started.md (hunk 1 above)
Comment (sredxny) [reply tyk-docs/content/tyk-identity-broker/getting-started.md]: done!

File: tyk-docs/content/tyk-identity-broker/getting-started.md (hunk 2 above)
Comment (dcs3spp) [reply tyk-docs/content/tyk-identity-broker/getting-started.md]: Ok thanks @sredxny

**Review by dcs3spp — APPROVED** [review]
Thanks @sredxny PR LGTM and approving. Once peer review and approval is complete, I'll merge ready for release.

File: tyk-docs/content/tyk-identity-broker/getting-started.md (hunk 2 above)
Comment (carolwkc) [inline tyk-docs/content/tyk-identity-broker/getting-started.md]:
```suggestion
### Configure secret for hashing session cookies
```

**Review by carolwkc — CHANGES_REQUESTED** [review]
Thanks @sredxny . Left a suggestion for the heading, otherwise the PR looks good.

File: tyk-docs/content/tyk-identity-broker/getting-started.md (hunk 3 above)
Comment (carolwkc) [reply tyk-docs/content/tyk-identity-broker/getting-started.md:42]: > TIB falls back to using the Tyk Dashboard's admin_secret when it's embedded in the dashboard.

@sredxny does this behaviour apply for 5.3+ only?

**Review by dcs3spp — COMMENTED** [review]
@sredxny If you are ok with suggestion made by @caroltyk could you commit the suggestion and then get approval from @caroltyk. Then we merge and release

**Review by dcs3spp — COMMENTED** [review]
@sredxny @caroltyk Should TYK_IB_SESSION_SECRET also be documented on this [page](https://tyk.io/docs/tyk-configuration-reference/tyk-identity-broker-configuration/#secret)?

**Comment by dcs3spp** [comment]
@caroltyk Would you be able to peer review?

**Review by carolwkc — APPROVED** [review]
lgtm

**Pull request merged**